### PR TITLE
Improve research catalysts

### PR DESCRIPTION
**Pull request by RusTit**
@mezz @Nedelosk please review.
But that's not all that needs to be done.
It is also necessary to register in botany.
https://github.com/ForestryMC/Binnie/blob/d503530df2762518559a0c97b5ab44e70dcb0146/botany-api/src/main/java/binnie/botany/api/BotanyAPI.java#L11
Currently there are no dependencies between them directly.
How should I do it right? Add a dependency or somehow programmatically check the load of the module? Or somehow?

### CODE REVIEW

**Comment by RusTit** [comment]
Related to https://github.com/ForestryMC/ForestryMC/issues/2034

**Comment by temp1011** [comment]
Use this to check if modules are loaded:
https://github.com/ForestryMC/ForestryMC/blob/2ce018d84b42518ed82a5a3e5fe374c88b325a9a/src/main/java/forestry/api/core/ForestryAPI.java#L90

**Comment by Nedelosk** [comment]
We only create a instance of a root if the module is enabled. So you only have to check if the value is null. If you have no access to the botany api you can get the root from the IAlleleRegistry with the uid of the root.
https://github.com/ForestryMC/ForestryMC/blob/2ce018d84b42518ed82a5a3e5fe374c88b325a9a/src/main/java/forestry/api/genetics/IAlleleRegistry.java#L37-L44
https://github.com/ForestryMC/Binnie/blob/d503530df2762518559a0c97b5ab44e70dcb0146/botany/src/main/java/binnie/botany/genetics/FlowerRoot.java#L46

**Comment by RusTit** [comment]
Please review.

File: extrabees/src/main/java/binnie/extrabees/items/ItemHoneyComb.java [hunk 1]
```diff
@@ -7,18 +7,35 @@
...
+	static final float RESEARCH_SUITABILITY = 0.4f;
+	static final String BOTANY_UID = "rootFlowers";
+
 	public ItemHoneyComb() {
 		super(EnumHoneyComb.values());
 		this.setCreativeTab(Tabs.tabApiculture);
 		this.setUnlocalizedName("honey_comb");
 		setRegistryName("honey_comb");
+		BeeManager.beeRoot.setResearchSuitability(new ItemStack(this, 1, OreDictionary.WILDCARD_VALUE), RESEARCH_SUITABILITY);
+		TreeManager.treeRoot.setResearchSuitability(new ItemStack(this, 1, OreDictionary.WILDCARD_VALUE), RESEARCH_SUITABILITY);
```
Comment (Nedelosk) [inline extrabees/src/main/java/binnie/extrabees/items/ItemHoneyComb.java]: `TreeManager.treeRoot` and `ButterflyManager.butterflyRoot` are null if the module that adds them is disabled. Please add null checks for both of them.

**Review by Nedelosk — CHANGES_REQUESTED** [review]

**Comment by RusTit** [comment]
Please review.

**Review by Nedelosk — APPROVED** [review]

**Pull request merged**